### PR TITLE
Added typescript type definitions

### LINKED ISSUE #109: Add type defininitions
Would be nice to have type definitions to use more easily in TS projects.

### PR DESCRIPTION
**Pull request by ntotten**
* Adds `src/get-jwks.d.ts`
* References types from `package.json`

Resolves #109

### CODE REVIEW

**Review by simoneb — COMMENTED** [review]
Thanks for this PR! Would you mind adding tests for type definitions using tsd?

File: src/get-jwks.d.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+type GetJwks = {
+  getPublicKey: (options?: {
+    domain?: string
+    alg?: string
+    kid?: string
+  }) => Promise<string>
+  getJwk: (signature: JWKSignature) => Promise<JWK>
+  getJwksUri: (normalizedDomain: string) => Promise<string>
+  catch: LRU<string, JWK>
```
Comment (radomird) [inline src/get-jwks.d.ts:14]: Shouldn't this be `cache` ?

**Review by radomird — CHANGES_REQUESTED** [review]

**Comment by radomird** [comment]
I agree with Simone, if you could add test that would be great. Thanks.

**Comment by simoneb** [comment]
@ntotten ping

**Comment by simoneb** [comment]
@radomird can we pick this up perhaps?

**Comment by radomird** [comment]
@simoneb & @ntotten 
Since I can't push anything to this branch: https://github.com/zuplo/get-jwks/tree/type_def I've created a fork, added tests there and opened a new PR: https://github.com/nearform/get-jwks/pull/115/ if that one is approved this one can be closed.

**Comment by radomird** [comment]
Closing this PR as https://github.com/nearform/get-jwks/pull/115 solved the issues mentioned in the comments above.

**Pull request closed without merging**